src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +309,127 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
+
+                    case "VmSwap":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out results.VmSwap);
+                        break;
+
+                    case "VmSize":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out results.VmSize);
+                        break;
+
+                    case "VmPeak":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out results.VmPeak);
+                        break;
+
+                    case "VmStk":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out ulong VmStk);
+                        // Add stack value to VmData to calculate memory usage
+                        results.VmData += VmStk;
+                        break;
+
+                    default:
+                        endIndex = statusFileContents.IndexOf('\n');
+                        break;
+                }
+                // Slice off the parsed part and parse on the rest.
+                statusFileContents = statusFileContents.Slice(endIndex + 1, statusFileContents.Length - endIndex - 1);
+            }
+            results.VmData *= 1024;
+            results.VmPeak *= 1024;
+            results.VmSize *= 1024;
+            results.VmSwap *= 1024;
+            results.VmRSS  *= 1024;
+            results.VmHWM  *= 1024;
+            result = results;
+            return true;
+        }
+
+        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, out string fileContents)
+        {
+            try
+            {
+                using (var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
+                {
+                    fileContents = reusableReader.ReadAllText(fileStream);
+                    return true;
+                }
+            }
+            catch (IOException)
+            {
+                fileContents = null;
+                return false;
+            }
+        }
+
+        private static bool TryReadFileAsSpan(string filePath, ReusableTextReader reusableReader, out ReadOnlySpan<char> fileContents)
+        {
+            try
+            {
+                using (var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
+                {
+                    fileContents = reusableReader.ReadAllText(fileStream).AsSpan();
+                    return true;
+                }
+            }
+            catch (IOException)
+            {
+                fileContents = null;
+                return false;
+            }
+        }
```

Comment:
This TryReadFileAsSpan function should be deleted.  It's identical to TryReadFile except that it's returning the result as a span.  The call site can instead use TryReadFile and do AsSpan on the result string to use it as a span.

Reply:
Done.